qiskit/primitives/sampler_result.py

```diff
@@ -30,8 +30,15 @@ class SamplerResult:
 
         result = sampler(circuits, params)
 
-    where the i-th elements of `result` correspond to the expectation using the circuit
-    given by `circuits[i]` and the parameters bounds by `params[i]`.
+    where the i-th element of ``result.quasi_dists`` corresponds to the probabilities
+    or the quasi-probabilities of bitstrings using the circuit given by ``circuits[i]``
+    and the parameters bounds by ``params[i]``.
+    The i-th element of ``result.metadata`` corresponds to the metadata of the experiment
+    using ``circuits[i]`` and ``params[i]``.
```

Comment:
I was intentionally vague here with `i-th elements of result` rather than `result.values[i]` because there might be several i-th elements eg `result.values`, `result.variances`, `result.confidence_interval`, `result.metadata` etc

Maybe something like:
```suggestion
    where the i-th elements of ``result`` corresponds to the circuit given by ``circuits[i]``,
    and the parameters bounds by ``params[i]``.
    For example, ``results.quasi_dists[i]`` gives the quasi-probabilities of bitstrings, and ``result.metadata[i]``
    is a metadata dictionary for this circuit and parameters.
```